### PR TITLE
Local Pluralities Get Free Xcm Execution

### PR DESCRIPTION
**Pull request by muharem**
Local Root and Pluralities Get Free Xcm Execution

After running e2e tests, we discovered that xcm programs sent from local pluralities such as Relay Chain Staking Admin or Collectives Fellowship were failing due to insufficient balance to cover execution/delivery fees. This PR addresses this issue by granting them a free execution.

Additionally, we encountered similar failures when attempting to teleport slashed assets from Collectives to the Relay Chain Treasury. In this PR, we resolve this problem by teleporting those assets on behalf of the Collectives root location, allowing for a free delivery.

### CODE REVIEW

File: system-parachains/collectives/collectives-polkadot/src/impls.rs [hunk 1]
```diff
@@ -38,36 +39,44 @@ pub type BalanceOf<T> =
...
 		let amount = match amount.drop_zero() {
 			Ok(..) => return,
 			Err(amount) => amount,
 		};
 		let imbalance = amount.peek();
-		let pallet_acc: AccountIdOf<T> = PalletAccount::get();
-		let treasury_acc: AccountIdOf<T> = TreasuryAccount::get();
-
-		<pallet_balances::Pallet<T>>::resolve_creating(&pallet_acc, amount);
+		let root_location: MultiLocation = Here.into();
```
Comment (muharem) [inline system-parachains/collectives/collectives-polkadot/src/impls.rs:62]: there are few options, 
- (current) resolve funds into root account temporarily and teleport on behalf of the root 
- build custom xcm program to teleport (do not use xcm pallet)
- keep as it is, and grant free execution to the pallet's account id. where a location of an account id has to be listed as `X1(AccountId32(some_custom_id))`

I think the current is the most optimal

**Comment by NachoPal** [comment]
New waived Origins works. However, it is failing with `Failed to convert root origin into account id` in the `on_unbalanced`.

**Comment by NachoPal** [comment]
Working now after the latest changes

File: relay/kusama/src/xcm_config.rs [hunk 2]
```diff
@@ -321,6 +325,10 @@ impl Contains<RuntimeCall> for SafeCallFilter {
 	}
 }
 
+/// Locations that will not be charged fees in the executor, either execution or delivery.
```
Comment (joepetrowski) [inline relay/kusama/src/xcm_config.rs:328]:
```suggestion
/// Locations that will not be charged fees in the executor, neither for execution nor delivery.
```

File: relay/polkadot/src/xcm_config.rs [hunk 3]
```diff
@@ -324,6 +328,10 @@ impl Contains<RuntimeCall> for SafeCallFilter {
 	}
 }
 
+/// Locations that will not be charged fees in the executor, either execution or delivery.
```
Comment (joepetrowski) [inline relay/polkadot/src/xcm_config.rs:331]:
```suggestion
/// Locations that will not be charged fees in the executor, neither for execution nor delivery.
```

**Review by joepetrowski — APPROVED** [review]

**Comment by bkchr** [comment]
@muharem could you please add some changelog entry?

**Comment by muharem** [comment]
> @muharem could you please add some changelog entry?

added info about new location to account id mapping, the waived locations type changes are basically setup for the new fee manager, we have a changelog regarding it in `Added` section, the changes to `ToParentTreasury` type does not change it's behaviour

**Review by bkchr — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

**Review by KiChjang — APPROVED** [review]

**Pull request merged**